examples/src/main/scala/io/github/rafafrdz/criteria4s/examples/BetweenExample.scala

```diff
@@ -0,0 +1,19 @@
+package io.github.rafafrdz.criteria4s.examples
+
+import io.github.rafafrdz.criteria4s.core._
+import io.github.rafafrdz.criteria4s.examples.datastores.{Postgres, WeirdDatastore}
+import io.github.rafafrdz.criteria4s.extensions._
+import io.github.rafafrdz.criteria4s.functions._
+
+object BetweenExample extends App {
+  val simpleBetween: Criteria[Postgres] = col[Postgres]("a").between(range(1, 10))
+  val simpleNotBetween: Criteria[Postgres] = col[Postgres]("b").notBetween(range("A", "Z"))
+
+  def reallyBoilerplatedCommonExpressionWithBetween[T <: CriteriaTag: BETWEEN: ({ type A[D <: CriteriaTag] = Show[Column, D] })#A: ({ type A[D <: CriteriaTag] = Show[(Int, Int), D] })#A]: Criteria[T] =
```

Comment:
I've never seen the "lambda types"construction expressions like that but now I get it. As you good say, that will be fixed by using the kind projections. We could use the [sbt plugin](https://github.com/typelevel/kind-projector) to get expressions like this:

`Show[Column, *]` instead of `({ type A[D <: CriteriaTag] = Show[Column, D] })#A`


```suggestion
  def reallyBoilerplatedCommonExpressionWithBetween[T <: CriteriaTag: BETWEEN: Show[Column, *]: Show[Seq[Int], *]]: Criteria[T] =
```